### PR TITLE
[MT-3568] EPCC Ref Store - Global search: Enable voice search

### PR DESCRIPTION
**Pull request by bdong8boss**
Description:
Add the voice search button next to the search box, when you click on the voice search, it has some animation to show it is listening for input. We use three icons for the voice search and it is added to the assets folder.

Linting:

- [x] No linting errors

Tests:

- [x] E2E tests
- [x] Manual tests
- [ ] Accessibility tests (no new react-axe errors in console)

Documentation:

- [ ] Requires documentation updates
- [ ] Requires Storybook component updates

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -9,6 +9,7 @@
     "@types/react-dom": "^16.9.0",
     "@types/react-instantsearch-dom": "^6.3.0",
     "@types/react-router-dom": "^5.1.5",
+    "@types/styled-components": "^5.1.2",
```
Comment (shaunmaharaj) [inline package.json:12]: Why do you need to pull in styled-components? We're not using them for our components. You should be able to use external saas files.

File: package.json [hunk 2]
```diff
@@ -28,6 +29,7 @@
     "react-scripts": "3.4.1",
     "serve": "^11.3.1",
     "sharethis-reactjs": "^1.5.0",
+    "styled-components": "^5.1.1",
```
Comment (shaunmaharaj) [inline package.json:32]: Is this intentional to pull two different versions?

File: src/SearchBar.scss [hunk 3]
```diff
@@ -80,6 +82,13 @@
       }
     }
 
+    .ais-SearchBox{
+      width: 100%;
+      width: -moz-available;          /* WebKit-based browsers will ignore this. */
```
Comment (dusanradovanovic) [inline src/SearchBar.scss:87]: Unless there is a specific reason for these, I would only keep `width: 100%` here. We already use postcss plugin which would add vendor specific prefixes for the less supported instructions although for width I think we only need 100%.

File: src/SearchBar.scss [hunk 4]
```diff
@@ -108,6 +117,11 @@
       padding-top: 3px;
     }
   }
+
+  .ais-VoiceSearch-status{
```
Comment (dusanradovanovic) [inline src/SearchBar.scss:121]: Missing space before {

File: src/SearchBar.scss [hunk 5]
```diff
@@ -108,6 +117,11 @@
...
+    display: none;
```
Comment (dusanradovanovic) [inline src/SearchBar.scss:122]: Is this used somewhere. `display: none` makes it permanently hidden.

File: src/SearchBar.tsx [hunk 6]
```diff
@@ -52,6 +109,25 @@ export const SearchBar: React.FC<SearchBoxProps> = () => {
...
   };
 
+  const VoiceSearchButtonText = ({
+    isListening,
+    isBrowserSupported
+  }: {isListening:any, isBrowserSupported:any}) => (
+    isBrowserSupported ? (
+      isListening ?
+        <SearchButtonMic
+          defaultIcon  ={Icon_SearchMicActive}
```
Comment (dusanradovanovic) [inline src/SearchBar.tsx:119]: Extra spaces. We do not use spaces around assignments in .jsx and .tsx files.

File: src/SearchBar.tsx [hunk 7]
```diff
@@ -52,6 +109,25 @@ export const SearchBar: React.FC<SearchBoxProps> = () => {
...
+          isListening  = {isListening}
```
Comment (dusanradovanovic) [inline src/SearchBar.tsx:120]: Extra spaces

File: src/SearchBar.tsx [hunk 8]
```diff
@@ -52,6 +109,25 @@ export const SearchBar: React.FC<SearchBoxProps> = () => {
...
+        /> :
+        <SearchButtonMic
+          isListening  = {isListening}
```
Comment (dusanradovanovic) [inline src/SearchBar.tsx:123]: Extra spaces

File: src/SearchBar.tsx [hunk 9]
```diff
@@ -52,6 +109,25 @@ export const SearchBar: React.FC<SearchBoxProps> = () => {
...
+          onClick={() => {
+            handleFocus()
```
Comment (dusanradovanovic) [inline src/SearchBar.tsx:125]: You can just write `onClick={() => handleFocus()}`

File: src/SearchBar.tsx [hunk 10]
```diff
@@ -88,6 +164,10 @@ export const SearchBar: React.FC<SearchBoxProps> = () => {
           reset={<ClearIcon />}
           translations={translations}
         />
+        <VoiceSearch
+          searchAsYouSpeak
```
Comment (dusanradovanovic) [inline src/SearchBar.tsx:168]: If this is a boolean prop, can you please set it to true explicitly?

**Comment by dusanradovanovic** [comment]
I've tried testing voice search on the Netlify instance but nothing happens? Google shows like it is recording sound for a second or two and than it just stops.

File: src/SearchBar.scss (hunk 5 above)
Comment (bdong8boss) [reply src/SearchBar.scss:122]: this status shows the info and status of the search, which is the default coming with the voicesearch widget. That's why we can only hide it.

File: src/SearchBar.tsx (hunk 10 above)
Comment (bdong8boss) [reply src/SearchBar.tsx:168]: yes it is a boolean prop.

**Comment by bdong8boss** [comment]
> I've tried testing voice search on the Netlify instance but nothing happens? Google shows like it is recording sound for a second or two and than it just stops.

It works for me. Could you please do a force refresh?

File: package.json (hunk 1 above)
Comment (bdong8boss) [reply package.json]: done!

File: package.json (hunk 2 above)
Comment (bdong8boss) [reply package.json]: removed.

File: src/SearchBar.scss (hunk 3 above)
Comment (bdong8boss) [reply src/SearchBar.scss]: fixed

File: src/SearchBar.scss [hunk 11]
```diff
@@ -107,7 +113,55 @@
...
+        background-image: url(./images/icons/icon-search-mic-active.svg);
+      }
+    }
+
+    @keyframes breatheAnimation {
+      100% { opacity: 1 }
+      50% { opacity: 0.65 }
+    }
+
+    .VoiceSearchButtonBreating {
```
Comment (shaunmaharaj) [inline src/SearchBar.scss:143]:
```suggestion
    .VoiceSearchButtonBreathing {
```

File: src/SearchBar.tsx [hunk 12]
```diff
@@ -13,6 +13,14 @@ import './SearchBar.scss';
 interface SearchBoxProps {
 }
 
+const SearchButtonMic = (props: any) => (
+  props.isListening ?
+    <span className="VoiceSearchButtonBreating"
```
Comment (shaunmaharaj) [inline src/SearchBar.tsx:18]:
```suggestion
    <span className="VoiceSearchButtonBreathing"
```

**Pull request merged**